sql/item_subselect.cc

```diff
@@ -4588,10 +4588,28 @@ void subselect_indexsubquery_engine::print(String *str,
   str->append(STRING_WITH_LEN("<index_lookup>("));
   tab->ref.items[0]->print(str, query_type);
   str->append(STRING_WITH_LEN(" in "));
-  str->append(tab->table->s->table_name.str, tab->table->s->table_name.length);
-  KEY *key_info= tab->table->key_info+ tab->ref.key;
-  str->append(STRING_WITH_LEN(" on "));
-  str->append(&key_info->name);
+  if (tab->table)
```

Comment:
Why can't we always get the TABLE* from `tab->tab_list->table` ?  I don't see where this won't be possible... 
(Perhaps, JOIN_TAB objects representing semi-join nests or aggregation nodes have tab->tab_list==NULL, but if we are in subselect_indexsubquery_engine or in a uniquesubquery_engine, it means the join had just one table and we are looking at the JOIN_TAB representing it... and not semi-join nest or aggregation node)